### PR TITLE
GH-1144 Amchart select persist 

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith Please Review ... if your ok with how i did this ill apply to the rest of amcharts 

Fixes GH-1144

### CODE REVIEW

File: src/amchart/Pie.js [hunk 1]
```diff
@@ -154,7 +167,13 @@
                     context._selected = {
                         field: field,
                         data: data
-                    };
+                    };
```
Comment (mlzummo) [inline src/amchart/Pie.js]: remove this block

File: src/amchart/Pie.js [hunk 2]
```diff
@@ -85,13 +88,20 @@
...
         } else {
-        	sortingMethod = function(a,b){ return a[1] > b[1] ? 1 : -1; };
+            sortingMethod = function(a,b){ return a[1] > b[1] ? 1 : -1; };
         }
         this.data(this.data().sort(sortingMethod));
 
         this._chart.colorField = "sliceColor";
-
-        this._chart.dataProvider = this.formatData(this.data());
+
```
Comment (mlzummo) [inline src/amchart/Pie.js]: do we want updates to persist after data has been changed (added or removed)

**Comment by mlzummo** [comment]
@GordonSmith Can you comment on this?

**Comment by GordonSmith** [comment]
Seems to be too much going on inside the setting of the color array.  I would suggest you need:
1.  A way to add/remove rows to this._selections, based on user interaction
2.  Inside the creation of the color map simply have a ```this._selections.indexOf(row) >= 0``` test.

**Comment by mlzummo** [comment]
@GordonSmith we will need to run .update on every click then ... also indexOf does a strict comparison so unless its a direct ref it wont work

**Comment by mlzummo** [comment]
running a rerender is going to be more expensive than computing a hash

**Comment by GordonSmith** [comment]
1.  If one of the chart elements has changed color due to a mouse action then you have to re-render?
2.  I was assuming your going to store the original row and not a clone, so indexof should work fine, if not, or if you need to persist the selection between complete data re-loads (which is not unlikely) then storing a hash of the row would be ideal (we will want to make sure we pick a suitable hashing algo. though).

**Comment by mlzummo** [comment]
@GordonSmith should selections persist even after a data update or should that invalidate a selection (if it invalidates it would make things so much easier)

**Comment by mlzummo** [comment]
closing in favor of: GH-1201

**Pull request closed without merging**